jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPartFormData.java

```diff
@@ -435,7 +429,13 @@ public void onPartContent(Content.Chunk chunk)
             if (fileName != null || isUseFilesForPartsWithoutFileName())
             {
                 long maxFileSize = getMaxFileSize();
-                fileSize += buffer.remaining();
+                long fileSize;
+                try (AutoLock ignored = lock.lock())
+                {
+                    fileSize = this.fileSize;
+                    fileSize += buffer.remaining();
+                    this.fileSize = fileSize;
```

Comment:
I find this way of doing it very confusing.  Isn't this just:
```suggestion
                    fileSize = this.fileSize += buffer.remaining();
```

or maybe:

```suggestion
                    this.fileSize += buffer.remaining();
                    fileSize = this.fileSize;
```

Reply:
Done.